File: api-tester/src/main/java/com/revenuecat/apitester/java/DangerousSettingsAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.DangerousSettings;

@SuppressWarnings({"unused"})
final class DangerousSettingsAPI {
    static void check(final DangerousSettings dangerousSettings) {
        final boolean autoSync = dangerousSettings.getAutoSyncPurchases();
        final boolean disableRequiredSignatureVerifications =
                dangerousSettings.getDisableRequiredSignatureVerifications();
        dangerousSettings.forceAllowTestStoreInReleaseBuilds();
    }

    static void checkConstructors() {
        final DangerousSettings defaults = new DangerousSettings();
        final DangerousSettings autoSync = new DangerousSettings(false);
        final DangerousSettings skipping = new DangerousSettings(true, true);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/DangerousSettingsAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.DangerousSettings

@Suppress("unused", "UNUSED_VARIABLE")
private class DangerousSettingsAPI {
    fun check(dangerousSettings: DangerousSettings) {
        val autoSync: Boolean = dangerousSettings.autoSyncPurchases
        val disableRequiredSignatureVerifications: Boolean =
            dangerousSettings.disableRequiredSignatureVerifications
        dangerousSettings.forceAllowTestStoreInReleaseBuilds()
    }

    fun checkConstructors() {
        val defaults = DangerousSettings()
        val autoSync = DangerousSettings(autoSyncPurchases = false)
        val skipping = DangerousSettings(
            autoSyncPurchases = true,
            disableRequiredSignatureVerifications = true,
        )
    }
}
